code/modules/events/tram_malfunction.dm

```diff
@@ -0,0 +1,62 @@
+#define TRAM_MALFUNCTION_TIME_UPPER 420
+#define TRAM_MALFUNCTION_TIME_LOWER 240
+#define XING_STATE_GREEN 0
+
+/datum/round_event_control/tram_malfunction
+	name = "Tram Malfunction"
+	typepath = /datum/round_event/tram_malfunction
+	weight = 20
+	max_occurrences = 4
+	earliest_start = 15 MINUTES
+	category = EVENT_CATEGORY_ENGINEERING
+
+//Check if there's a tram we can cause to malfunction.
+/datum/round_event_control/tram_malfunction/can_spawn_event(players_amt)
+	. = ..()
```

Comment:
Might want to add a check for if this is false:

```suggestion
	. = ..()
	if (!.)
		return FALSE

```

Reply:
Yeah I think I had that earlier. I'll test this all out.